proxy/src/test/java/com/velocitypowered/proxy/scheduler/VelocitySchedulerTest.java

```diff
@@ -65,4 +66,39 @@ void repeatTaskWorks() throws Exception {
     task.cancel();
   }
 
+  @Test
+  void nullSchedulerTest() throws Exception {
+    VelocityScheduler scheduler = new VelocityScheduler(new FakePluginManager());
+
+    assertThrows(NullPointerException.class, () -> scheduler.builder(null).schedule());
+    assertThrows(IllegalArgumentException.class, () -> scheduler.builder(FakePluginManager.PLUGIN_A)
+        .schedule());
+  }
+
+  @Test
+  void obtainTasksFromPlugin() throws Exception {
+    VelocityScheduler scheduler = new VelocityScheduler(new FakePluginManager());
+
+    scheduler.builder(FakePluginManager.PLUGIN_A)
+      .task(ScheduledTask::cancel)
+      .delay(100, TimeUnit.MILLISECONDS)
+      .schedule();
+
+    assertEquals(scheduler.tasksByPlugin(FakePluginManager.PLUGIN_A).size(), 1);
+  }
+
+  @Test
+  void testConsumerCancel() throws Exception {
+    VelocityScheduler scheduler = new VelocityScheduler(new FakePluginManager());
+
+    ScheduledTask task = scheduler.builder(FakePluginManager.PLUGIN_B)
+        .task(ScheduledTask::cancel)
```

Comment:
Same as above: we could schedule a task inside the original task that checks its status.